grid_map_core/include/grid_map_core/GridMap.hpp

```diff
@@ -514,7 +514,7 @@ class GridMap
   Time timestamp_;
 
   //! Grid map data stored as layers of matrices.
-  std::unordered_map<std::string, Matrix> data_;
+  std::unordered_map<std::string, Matrix, std::hash<std::string>, std::equal_to<std::string>, Eigen::aligned_allocator<std::pair<std::string, Matrix> > > data_;
```

Comment:
@ThomasEmter I am trying to compile this branch with clang, and I am getting errors similar to https://github.com/JakobEngel/dso/issues/111
Following the same fixes in ^, I added `const` before `std::string` in the line above and passed clang compilation. You may want to change it here. 
I did not investigate the implications of using a const string except the fact that it compiles with clang. 
Gcc is fine in both ways.

Reply:
@MirkoFerrati thank you for looking into the pull request. You are absolutely right, the `std::string` has to be be `const` according to [Eigen docs](https://eigen.tuxfamily.org/dox/group__TopicStlContainers.htm). I must have missed that and I indeed only used GCC, so I did not get the compile errors.
Should be fixed now.